### PR TITLE
Implemented Registry pattern

### PR DESCRIPTION
**Pull request by viveksb007**
Registry pattern

- Implemented Registry pattern addressing the issue #1310 

Description

- Added registry submodule and implemented the pattern there.
- Example for registry pattern was discussed in the issue itself #1310

### CODE REVIEW

**Comment by viveksb007** [comment]
#1538 will fix the build for this PR too.

**Comment by viveksb007** [comment]
@iluwatar can you have a look?
Thanks

**Comment by ravening** [comment]
@viveksb007 Registry pattern is best acheived factory pattern so that you instantiate objects using factories which stores the created objects in registry and using registry you can fetch objects.

Also if you use factory, you can instantiate not just customers but other object types as well. It provides better way to handle errors as well

**Comment by viveksb007** [comment]
@ravening thanks for looking. I get that its better to initialise object through factory and storing in registry then and there. But I actually discussed before implementation here #1310 with @iluwatar. It was concluded not to involve factory and keep the registry implementation simple.

**Comment by ravening** [comment]
> @ravening thanks for reviewing. I get that its better to initialise object through factory and storing in registry then and there. But I actually discussed before implementation here #1310 with @iluwatar. It was concluded not to involve factory and keep the registry implementation simple. 

@viveksb007 ok

**Review by ohbus — APPROVED** [review]
A very good PR indeed.

Thanks @viveksb007 for this awesome design.

File: registry/etc/registry.urm.puml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    + getName() : String
+    + toString() : String
+  }
+  class CustomerRegistry {
+    + addCustomer(customer : Customer)
+    + getCustomer(id : String)
+  }
+}
+Customer --> "-addCustomer" CustomerRegistry
+@enduml
```
[the remark this answers is not in the document]
Comment (viveksb007) [reply registry/etc/registry.urm.puml]: Done

**Comment by viveksb007** [comment]
@ohbus thanks a lot for the review and feedback.

**Review by ohbus — APPROVED** [review]

**Comment by ohbus** [comment]
Thank you @viveksb007 for this awesome contribution.

@all-contributors please add @viveksb007 for code

**Pull request merged**